django/src/rdwatch_scoring/views.py

```diff
@@ -0,0 +1,216 @@
+
+from .models import (
+    EvaluationActivityClassificationTemporalIou,
+    EvaluationBroadAreaSearchDetection,
+    EvaluationRun,
+    Site,
+)
+
+logger = logging.getLogger(__name__)
+
+
+router = RouterPaginated()
+
+
+# Query for the Scoring Database to get unique truth typs based on the parmeters
+def get_unique_site_truth_types(evaluation_run_uid, activity_type, generatedSiteId):
+    site_proposals = EvaluationBroadAreaSearchDetection.objects.filter(
+        evaluation_run_uuid=evaluation_run_uid,
+        activity_type=activity_type,
+        site_proposal_matched__contains=generatedSiteId,
+    ).distinct()
+
+    # Retrieve unique site_truth_type values from the filtered site proposals
+    unique_site_truth_types = site_proposals.values_list(
+        'site_truth_type', flat=True
+    ).distinct()
+
+    return list(unique_site_truth_types)
+
+
+# Gets coloring for sites based on their scoring results
+def get_site_scoring_color(evaluation_run_uid, activity_type, generatedSiteId):
+    match_statuses = get_unique_site_truth_types(
+        evaluation_run_uid, activity_type, generatedSiteId
+    )
+    sm_color = 'magenta'
+    if len(match_statuses):
+        if set(match_statuses) & {
+            'positive_annotated',
+            'positive_annotated_static',
+            'positive_partial',
+            'positive_pending',
+        }:
+            # and at least 1 negative match (partially wrong)
+            if set(match_statuses) & {
+                'positive_excluded',
+                'negative',
+                'negative_unbounded',
+            }:
+                sm_color = 'orange'
+            else:
+                sm_color = '#66CCAA'
+        # only negative matches (completely wrong)
+        elif set(match_statuses) & {
+            'positive_excluded',
+            'negative',
+            'negative_unbounded',
+        }:
+            sm_color = 'magenta'
+        elif set(match_statuses) & {'ignore'}:
+            sm_color = 'lightsalmon'
+    return sm_color
+
+
+class TemporalSchema(Schema):
+    site_preparation: str = None
+    active_construction: str = None
+    post_construction: str = None
+
+
+class EvaluationResponseSchema(Schema):
+    region_name: str = None
+    evaluationId: int = None
+    evaluation_run: int = None
+    performer: str = None
+    evaluation_uuid: str = None
+    unionArea: float = None
+    statusAnnotated: str = None
+    temporalIOU: TemporalSchema = None
+    color: str = None
+
+
+# If the combination of configurationId and regionId has scoring
+@router.get('/has_scores')
+def has_scores(request: HttpRequest, configurationId: int, regionId: int):
+    # from the hyper parameters we need the evaluation and evaluation_run Ids
+    configuration = get_object_or_404(HyperParameters, pk=configurationId)
+    evaluationId = configuration.evaluation
+    performer_name = configuration.performer.slug.lower()
+    evaluation_run = configuration.evaluation_run
```

Comment:
The ORM will generate two SQL queries here, when we can do it in one with [`select_related`](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#select-related):
```suggestion
    configuration = get_object_or_404(HyperParameters.objects.select_related('performer'), pk=configurationId)
    evaluationId = configuration.evaluation
    performer_name = configuration.performer.slug.lower()
    evaluation_run = configuration.evaluation_run
```